sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs

```diff
@@ -428,10 +428,15 @@ private void InstrumentMessages(IEnumerable<ServiceBusMessage> messages)
             DateTimeOffset scheduledEnqueueTime,
             CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNullOrEmpty(messages, nameof(messages));
+            Argument.AssertNotNull(messages, nameof(messages));
             Argument.AssertNotDisposed(IsClosed, nameof(ServiceBusSender));
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
             var messageList = messages.ToList();
```

Comment:
`ToList` will always allocate.   You may want to give it a type sniff if you don't need a new set.  Event Hubs does something similar to:

```csharp
var messageList = messages switch
{
    IReadOnlyList<ServiceBusMessage> alreadyAList => alreadyAList ,
    _ => messages .ToList()
};
```

Reply:
Why check specifically for IReadOnlyList as opposed to IList?

Reply:
In our case, we wanted to make a copy if the set passed in was mutable.

Reply:
That is interesting - when would you expect customers to actually pass in a read only list here?

Reply:
I'm wrong;  our lower level expects `IReadOnlyList` and we can't be sure that something with `IList` supports it.    _(List supports both, so the mutable rationale was my mistake)_

Reply:
I see, but still this is only helpful if users actually pass in an IReadOnlyList which means they likely already copied something. Why not just pass in the thing that they were copying directly?

Reply:
It will also catch anything that can be cast to `IReadOnlyList<T>`,  such as  `List<T>` - which is pretty common to use for `IEnumerable<T>`

Reply:
Ah ok I didn't realize it would also catch List<T>. Thanks!

Reply:
I would expect this change to show some nice perf gains in our perf testing /cc @MiYanni